db/db_impl.cc

```diff
@@ -646,8 +646,21 @@ void DBImpl::StartTimedTasks() {
   }
 }
 
+size_t DBImpl::GetStatsHistorySize() const {
+  if (stats_history_.empty()) return 0;
+  size_t size_total = sizeof(stats_history_);
+  for (const auto& stats : stats_history_) {
+    size_total += sizeof(stats.first) + sizeof(stats.second);
+    if (stats.second.size() != 0) {
+      auto s = stats.second.begin();
+      size_total += (sizeof(s->first) + sizeof(s->second)) * stats.second.size();
```

Comment:
Here we go though every time slice, and for each, estimating the memory by using the first one in the map, times map size. I think a much more accurate estimation will be the other around: do a more accurate estimate of the memory consumption for one time slice, and times number of time slices.